static/app/views/settings/organizationProjects/index.tsx

```diff
@@ -27,113 +31,106 @@ import ProjectStatsGraph from './projectStatsGraph';
-          <PanelBody>
-            {projectList ? (
-              sortProjects(projectList).map(project => (
-                <GridPanelItem key={project.id}>
-                  <ProjectListItemWrapper>
-                    <ProjectListItem project={project} organization={organization} />
-                  </ProjectListItemWrapper>
-                  <ProjectStatsGraphWrapper>
-                    {projectStats ? (
-                      <ProjectStatsGraph
-                        key={project.id}
-                        project={project}
-                        stats={projectStats[project.id]}
-                      />
-                    ) : (
-                      <Placeholder height="25px" />
-                    )}
-                  </ProjectStatsGraphWrapper>
-                </GridPanelItem>
-              ))
-            ) : (
-              <LoadingIndicator />
-            )}
-            {projectList && projectList.length === 0 && (
-              <EmptyMessage>{t('No projects found.')}</EmptyMessage>
-            )}
-          </PanelBody>
-        </Panel>
-        {projectListPageLinks && (
-          <Pagination pageLinks={projectListPageLinks} {...this.props} />
-        )}
-      </Fragment>
-    );
-  }
+      },
+    ],
+    {staleTime: Infinity}
+  );
+
+  const projectListPageLinks = getResponseHeader?.('Link');
+
+  const action = <CreateProjectButton />;
+
+  const debouncedSearch = debounce(
+    (searchQuery: string) =>
+      browserHistory.replace({
+        pathname: location.pathname,
+        query: {...location.query, query: searchQuery},
+      }),
+    DEFAULT_DEBOUNCE_DURATION
+  );
+
+  return (
+    <Fragment>
+      <SentryDocumentTitle
+        title={routeTitleGen(t('Projects'), organization.slug, false)}
+      />
+      <SettingsPageHeader title="Projects" action={action} />
+      <SearchWrapper>
+        <SearchBar
+          placeholder={t('Search Projects')}
+          onChange={debouncedSearch}
+          query={query}
+        />
+      </SearchWrapper>
+      <Panel>
+        <PanelHeader>{t('Projects')}</PanelHeader>
+        <PanelBody>
+          {projectList ? (
+            sortProjects(projectList).map(project => (
+              <GridPanelItem key={project.id}>
+                <ProjectListItemWrapper>
+                  <ProjectListItem project={project} organization={organization} />
+                </ProjectListItemWrapper>
+                <ProjectStatsGraphWrapper>
+                  {projectStats ? (
+                    <ProjectStatsGraph
+                      key={project.id}
+                      project={project}
+                      stats={projectStats[project.id]}
+                    />
+                  ) : (
+                    <Placeholder height="25px" />
+                  )}
+                </ProjectStatsGraphWrapper>
+              </GridPanelItem>
+            ))
+          ) : (
+            <LoadingIndicator />
```

Comment:
This is how it used to work, but we can do a little better now. Let's add `isLoading` and `isError` so you can render `LoadingError` in case of an error and only show the loading state when it's actually loading.